app/src/main/kotlin/co/netguru/baby/monitor/client/feature/noisedetection/NoiseDetector.kt

```diff
@@ -10,22 +11,27 @@ import kotlin.math.sqrt
 
 class NoiseDetector @Inject constructor() {
     fun processData(data: ShortArray) = Single.fromCallable<Int> {
-        var decibels = 0.0
-        if (data.isNotEmpty()) {
-            var totalSquared = 0.0
-            for (soundBit in data) {
-                totalSquared += (soundBit * soundBit).toDouble()
-            }
+        val maxDecibels = data.asSequence()
+            .chunked(ANALYZE_DATA_CHUNK_SIZE)
+            .map(this::mapToDecibels)
+            .max()
 
-            val quadraticMeanPressure = sqrt(totalSquared / data.size)
-            decibels = FORMULA_CONSTANT * log10(quadraticMeanPressure)
-        }
-        Timber.d("Sound Db: $decibels")
-        return@fromCallable decibels.roundToInt()
+        Timber.d("Max Db: $maxDecibels")
+        return@fromCallable maxDecibels?.roundToInt() ?: 0
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/netguru/baby-monitor-client-android/blob/7c069c26782df05dac8a097aa956e351ad95f920/app/src/main/kotlin/co/netguru/baby/monitor/client/feature/noisedetection/NoiseDetector.kt#L20"></span>Expression with labels increase complexity and affect maintainability.</td>
    </tr>
  </tbody>
</table>